pandas/core/frame.py

```diff
@@ -293,10 +293,17 @@
 
       .. versionadded:: 1.2.0
 
-on : label or list
+on : label or list, or Callable
     Column or index level names to join on. These must be found in both
     DataFrames. If `on` is None and not merging on indexes then this defaults
     to the intersection of the columns in both DataFrames.
+    If Callable this will perform a conditional merge. This allows you to
```

Comment:
Looks like this description still needs to talk about the default chunk size and how to change it.